pyart/io/output_to_geotiff.py

```
@@ -0,0 +1,246 @@
+            out_driver = gdal.GetDriverByName("GTiff")
+
+            # Output dataset depends on rgb flag
+            if not rgb:
+                # Single-channel, floating-point output
+                dst_options = ['COMPRESS=LZW', 'ALPHA=YES']
+                dst_ds = out_driver.Create(
+                    ofile, int(2*rcells), int(2*rcells), 1,
+                    gdal.GDT_Float32, dst_options)
+            else:
+                # Assign data RGB levels based on value relative to vmax/vmin
+                rarr, garr, barr = _get_rgb_values(
+                    data, vmin, vmax, color_levels, cmap)
+                dst_ds = out_driver.Create(ofile, data.shape[1],
+                                           data.shape[0], 3, gdal.GDT_Byte)
+
+            # Common Projection and GeoTransform
+            dst_ds.SetGeoTransform([-dist, -rangestep, 0, dist, 0, rangestep])
+            dst_ds.SetProjection(iproj)
+
+            # Final output depends on rgb flag
+            if not rgb:
+                dst_ds.GetRasterBand(1).WriteArray(data[::-1, :])
+            else:
+                dst_ds.GetRasterBand(1).WriteArray(rarr[::-1, :])
+                dst_ds.GetRasterBand(2).WriteArray(garr[::-1, :])
+                dst_ds.GetRasterBand(3).WriteArray(barr[::-1, :])
+            dst_ds.FlushCache()
+            dst_ds = None
+
+            if sld:
+                _create_sld(cmap, vmin, vmax, ofile, color_levels)
+
+            if warp:
+                # Warps TIFF to lat/lon WGS84 projection that is more useful
+                # for web mapping applications. Likely changes array shape.
+                os.system('gdalwarp -t_srs \'+proj=longlat +ellps=WGS84 ' +
+                          '+datum=WGS84 +no_defs\' ' + ofile + ' ' +
+                          ofile + '_tmp.tif')
+                os.system('mv ' + ofile + '_tmp.tif' + ' ' + ofile)
+    if not f_flag:
+        print('Failed - field not found.')
```

This should raise an exception.